contracts/governance/StakingRewards/StakingRewards.sol

```diff
@@ -72,14 +59,59 @@ contract StakingRewards is StakingRewardsStorage, Initializable {
 	 * after intervals of 14 days.
 	 * */
 	function collectReward() external {
-		uint256 withdrawalTime;
-		uint256 amount;
-		(withdrawalTime, amount) = getStakerCurrentReward(true);
+		(uint256 withdrawalTime, uint256 amount) = getStakerCurrentReward(true);
 		require(withdrawalTime > 0 && amount > 0, "no valid reward");
 		withdrawals[msg.sender] = withdrawalTime;
 		_payReward(msg.sender, amount);
 	}
 
+	/**
+	 * @notice Withdraws all token from the contract by Multisig.
+	 * @param _receiverAddress The address where the tokens has to be transferred.
+	 */
+	function withdrawTokensByOwner(address _receiverAddress) external onlyOwner {
+		uint256 value = SOV.balanceOf(address(this));
+		_transferSOV(_receiverAddress, value);
+	}
+
+	/**
+	 * @notice Changes average block time - based on blockchain
+	 * @dev If average block time significantly changes, we can update it here and use for block number calculation
+	 */
+	function setAverageBlockTime(uint256 _averageBlockTime) external onlyOwner {
+		averageBlockTime = _averageBlockTime;
+	}
+
+	/**
+	 * @notice This function computes the last staking checkpoint and calculates the corresponding
+	 * block number using the average block time which is then added to the mapping `checkpointBlockDetails`.
+	 */
+	function setBlock() external {
+		uint256 lastCheckpointTime = staking.timestampToLockDate(block.timestamp);
+		_setBlock(lastCheckpointTime);
+	}
+
+	/**
+	 * @notice This function computes the block number using the average block time for a given historical
+	 * checkpoint which is added to the mapping `checkpointBlockDetails`.
+	 * @param _time Exact staking checkpoint time
+	 */
+	function setHistoricalBlock(uint256 _time) external onlyOwner {
+		_setBlock(_time);
+	}
```

Comment:
if for some reason time between blocks will change several times (e.g. Ethereum's difficulty bomb), this trick might not work well, especially for older blocks. so this contract requires regular maintenance.